src/twisted/conch/scripts/ckeygen.py

```diff
@@ -23,13 +24,25 @@
 
 
 
+supportedKeyTypes = dict()
+def _keyGenerator(keyType):
+    def assignkeygenerator(keygenerator):
+        supportedKeyTypes[keyType] = keygenerator
+        @wraps(keygenerator)
+        def wrapper(*args, **kwargs):
+            keygenerator(*args, **kwargs)
+        return wrapper
```

Comment:
decorators with arguments works a little differently http://stackoverflow.com/questions/5929107/python-decorators-with-parameters, but I am quite why those lines are not covered I will try to figure this out tomorrow.